crypto/src/main/java/net/corda/v5/crypto/SecureHash.java

```diff
@@ -0,0 +1,99 @@
+package net.corda.v5.crypto;
+
+import net.corda.v5.base.annotations.CordaSerializable;
+import net.corda.v5.base.types.ByteArrays;
+import net.corda.v5.base.types.OpaqueBytes;
+import org.jetbrains.annotations.NotNull;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Container for a cryptographically secure hash value.
+ * Provides utilities for generating a cryptographic hash using different algorithms (currently only SHA-256 supported).
+ */
+@CordaSerializable
+public final class SecureHash extends OpaqueBytes {
+    /**
+     * Construct a secure hash.
+     *
+     * @param algorithm Hashing algorithm which was used to generate the hash.
+     * @param bytes     Hash value.
+     */
+    public SecureHash(@NotNull String algorithm, @NotNull byte[] bytes) {
+        super(bytes);
+        this.algorithm = algorithm;
+    }
+
+    static final char DELIMITER = ':';
+
+    /**
+     * Creates a [SecureHash].
+     * <p>
+     * This function does not validate the length of the created digest. Prefer using
+     * [net.corda.v5.application.crypto.HashingService.parse] for a safer mechanism for creating [SecureHash]es.
+     *
+     */
+
+    public static SecureHash parse(String str) {
+        int idx = str.indexOf(DELIMITER);
+        if (idx == -1) {
+            throw new IllegalArgumentException("Provided string: $str should be of format algorithm:hexadecimal");
+        } else {
+            String algorithm = str.substring(0, idx);
+            String value = str.substring(idx + 1);
+            byte[] data = ByteArrays.parseAsHex(value);
+            return new SecureHash(algorithm, data);
+        }
+    }
+
+    @NotNull
+    public final String algorithm;
```

Comment:
I think this needs to become private and add a getter for it

Reply:
done